### DISCUSSION TITLE
Correct max and min size for 32 bit integer?

### DISCUSSION THREAD

**Discussion by TheFern2** [micropython/micropython#10315] [Unix & Windows] [answered]
For python3 that is -2147483648 through 2147483647. However -2147483648 causes an overflow in upy. Both on unix and esp32 ports.

I decrease to -2147483647 and that fixed my overflow issue.

I was using it in the contest of:
works in py2, py3, but overflow in upy.
```
random.randint(-2147483648, 2147483647)
```

In unix port my unittest that was expecting the randint was just stuck with no exceptions or errors probably a port bug. On esp32 1.19 I was getting the overflow error and that's how I found the issue.

**Comment by peterhinch** [comment, marked as answer]
Python integers are unbounded. At the C level a MicroPython small int is 30 bits.
